torch/fx/experimental/symbolic_shapes.py

```diff
@@ -6890,6 +6897,34 @@ def compute_concrete_val() -> sympy.Basic:
                         concrete_val = unsound_result
                         ok = True
 
+                    if not ok and self.trace_asserts:
+                        # check if this boolean is used in an assertion, bytecode pattern for assertions
+                        # is pretty stable for Python 3.7--3.12, ported with minimal changes from
+                        # torch/fx/proxy.py
+                        frame = self._get_user_frame()
+                        assert frame is not None
+
+                        insts = list(dis.get_instructions(frame.f_code))
+                        if sys.version_info >= (3, 11):
+                            from bisect import bisect_left
+                            cur = bisect_left(insts, frame.f_lasti, key=lambda x: x.offset)
+                        else:
+                            cur = frame.f_lasti // 2
+                        inst = insts[cur]
+
+                        if inst.opname == "POP_JUMP_IF_TRUE" and inst.arg is not None:
+                            first = insts[cur + 1]
+
+                            starts_with_assert = (
+                                first.opname == "LOAD_GLOBAL"
+                                and first.argval == "AssertionError"
+                                or first.opname == "LOAD_ASSERTION_ERROR"
+                            )
+                            if starts_with_assert and insts[cur + 2].opname == "RAISE_VARARGS":
+                                concrete_val = sympy.true
+                                transmute_into_runtime_assert = True
```

Comment:
For my understanding: where does it actually add torch._check in symbolic tracing? I thought you were supposed to do sth like https://github.com/pytorch/pytorch/pull/120564/files#diff-622913fdb49db90d6f3a8ab225b4badb7996023e6498e9f7c6d03fe9f32d0986R351

Reply:
I think there might be recursion issue there since we are already lower in the stack here, we can do that as BE later.